src/main/java/it/gov/pagopa/pu/workflow/wf/classification/iuf/classification/IufClassificationWFImpl.java

```diff
@@ -0,0 +1,136 @@
+package it.gov.pagopa.pu.workflow.wf.classification.iuf.classification;
+
+import io.temporal.spring.boot.WorkflowImpl;
+import it.gov.pagopa.payhub.activities.activity.classifications.*;
+import it.gov.pagopa.payhub.activities.dto.classifications.IufClassificationActivityResult;
+import it.gov.pagopa.payhub.activities.dto.classifications.Transfer2ClassifyDTO;
+import it.gov.pagopa.pu.workflow.wf.classification.iuf.dto.IufClassificationNotifyPaymentsReportingSignalDTO;
+import it.gov.pagopa.pu.workflow.wf.classification.iuf.dto.IufClassificationNotifyTreasurySignalDTO;
+import it.gov.pagopa.pu.workflow.wf.classification.iuf.config.IufClassificationWfConfig;
+import it.gov.pagopa.pu.workflow.wf.classification.iuf.activity.StartTransferClassificationActivity;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static it.gov.pagopa.pu.workflow.wf.ingestionflow.paymentsreporting.wfingestion.PaymentsReportingIngestionWFImpl.TASK_QUEUE;
+
+/**
+ * Implementation of the IufClassificationWF interface
+ */
+@Slf4j
+@WorkflowImpl(taskQueues = TASK_QUEUE)
+public class IufClassificationWFImpl implements IufClassificationWF, ApplicationContextAware {
+  public static final String TASK_QUEUE = "IufClassificationWF";
+
+  private IufClassificationNotifyTreasurySignalDTO signalDTO;
+
+  private ClearClassifyIufActivity clearClassifyIufActivity;
+  private IufClassificationActivity iufClassificationActivity;
+  private TransferClassificationActivity transferClassificationActivity;
+
+  private StartTransferClassificationActivity startTransferClassificationActivity;
+
+  private IufClassificationActivityResult result;
```

Comment:
let's init with an empty collection, the signal will just add a new element, the workflowMethod will handle the flattening of the nested request
```suggestion
  private List<IufClassificationActivityResult> toNotify = new ArrayList<>();
```

Reply:
OK